File: classical-components-demo/beta-barcode-scanner/src/main/java/io/scanbot/example/BarcodeTypesAdapter.java
package io.scanbot.example;

import android.util.SparseArray;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;
import android.widget.CheckBox;
import android.widget.TextView;

import androidx.annotation.NonNull;
import androidx.recyclerview.widget.RecyclerView;

import java.util.ArrayList;

import io.scanbot.sdk.barcode.entity.BarcodeFormat;

public class BarcodeTypesAdapter extends RecyclerView.Adapter<BarcodeTypesAdapter.BarcodeTypesViewHolder> {

    public final SparseArray<BarcodeFormat> selectedFormats = new SparseArray<>();

    private final ArrayList<Integer> preSelectedBarcodeFormatsIndexes;

    public BarcodeTypesAdapter(ArrayList<Integer> preSelectedBarcodeFormatsIndexes) {
        this.preSelectedBarcodeFormatsIndexes = preSelectedBarcodeFormatsIndexes;
    }

    public static class BarcodeTypesViewHolder extends RecyclerView.ViewHolder {
        public TextView barcodeTypeName;
        public CheckBox barcodeTypeChecker;

        public BarcodeTypesViewHolder(@NonNull View barcodeTypeView) {
            super(barcodeTypeView);
            this.barcodeTypeName = barcodeTypeView.findViewById(R.id.barcode_type_name);
            this.barcodeTypeChecker = barcodeTypeView.findViewById(R.id.barcode_type_checker);
        }
    }

    @Override
    public BarcodeTypesViewHolder onCreateViewHolder(ViewGroup parent,
                                                     int viewType) {
        View barcodeTypeView = LayoutInflater.from(parent.getContext())
                .inflate(R.layout.barcode_type_view, parent, false);

        BarcodeTypesViewHolder viewHolder = new BarcodeTypesViewHolder(barcodeTypeView);
        return viewHolder;
    }

    @Override
    public void onBindViewHolder(BarcodeTypesViewHolder holder, int position) {
        BarcodeFormat barcodeFormat = BarcodeFormat.values()[position];
        holder.barcodeTypeName.setText(barcodeFormat.name());
        holder.barcodeTypeChecker.setOnCheckedChangeListener((buttonView, isChecked) -> {
            if (isChecked) {
                selectedFormats.append(position, barcodeFormat);
            } else {
                selectedFormats.remove(position);
            }
        });

        if (preSelectedBarcodeFormatsIndexes.contains(barcodeFormat.ordinal())) {
            holder.barcodeTypeChecker.setChecked(true);
        }
    }

    @Override
    public int getItemCount() {
        return BarcodeFormat.values().length;
    }
}

File: classical-components-demo/barcode-scanner/src/main/java/io/scanbot/example/ui/adapter/BarcodeTypesAdapter.kt
package io.scanbot.example.ui.adapter

import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import android.widget.CheckBox
import android.widget.TextView
import androidx.recyclerview.widget.RecyclerView
import io.scanbot.example.R
import io.scanbot.example.repository.BarcodeTypeRepository
import io.scanbot.example.ui.adapter.BarcodeTypesAdapter.BarcodeTypesViewHolder
import io.scanbot.sdk.barcode.entity.BarcodeFormat

class BarcodeTypesAdapter() :
    RecyclerView.Adapter<BarcodeTypesViewHolder>() {
    val values = BarcodeFormat.values()


    class BarcodeTypesViewHolder(barcodeTypeView: View) : RecyclerView.ViewHolder(barcodeTypeView) {
        var barcodeTypeName: TextView
        var barcodeTypeChecker: CheckBox

        init {
            this.barcodeTypeName = barcodeTypeView.findViewById(R.id.barcode_type_name)
            this.barcodeTypeChecker = barcodeTypeView.findViewById(R.id.barcode_type_checker)
        }
    }

    override fun onCreateViewHolder(
        parent: ViewGroup,
        viewType: Int
    ): BarcodeTypesViewHolder {
        val barcodeTypeView = LayoutInflater.from(parent.context)
            .inflate(R.layout.barcode_type_view, parent, false)

        return BarcodeTypesViewHolder(barcodeTypeView)
    }

    override fun onBindViewHolder(holder: BarcodeTypesViewHolder, position: Int) {
        val barcodeFormat = values[position]
        holder.barcodeTypeName.text = barcodeFormat.name
        holder.barcodeTypeChecker.setOnCheckedChangeListener { buttonView, isChecked ->
            if (isChecked) {
                BarcodeTypeRepository.selectType(barcodeFormat)
            } else {
                BarcodeTypeRepository.deselectType(barcodeFormat)
            }
        }

        if (BarcodeTypeRepository.selectedTypes.contains(barcodeFormat)) {
            holder.barcodeTypeChecker.isChecked = true
        }
    }

    override fun getItemCount(): Int {
        return values.size
    }
}
